android/src/main/java/com/dooboolab/fluttersound/FlutterSoundPlugin.java

```diff
@@ -134,43 +210,43 @@ public boolean onRequestPermissionsResult(int requestCode, String[] permissions,
-                  , 0 // CODEC_MP3 (not implemented)
-                  , MediaRecorder.AudioEncoder.VORBIS
-                  , 0 // CODEC_PCM (not implemented)
-          };
-  int formatsArray[] = {
-                  MediaRecorder.OutputFormat.MPEG_4 // DEFAULT
-                  , MediaRecorder.OutputFormat.MPEG_4 // CODEC_AAC
-                  , MediaRecorder.OutputFormat.OGG    // CODEC_OPUS
-                  , 0                                 // CODEC_CAF_OPUS (this is apple specific)
-                  , 0                                 // CODEC_MP3
-                  , MediaRecorder.OutputFormat.OGG    // CODEC_VORBIS
-                  , 0                                 // CODEC_PCM
-
-          };
-
-
   @Override
   public void startRecorder(Integer numChannels, Integer sampleRate, Integer bitRate, t_CODEC codec, int androidEncoder, int androidAudioSource, int androidOutputFormat, String path, final Result result) {
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-      if (
-          reg.activity().checkSelfPermission(Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED
-              || reg.activity().checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED
-          ) {
-        reg.activity().requestPermissions(new String[]{
-            Manifest.permission.RECORD_AUDIO,
-            Manifest.permission.WRITE_EXTERNAL_STORAGE,
-        }, 0);
-        result.error(TAG, "NO PERMISSION GRANTED", Manifest.permission.RECORD_AUDIO + " or " + Manifest.permission.WRITE_EXTERNAL_STORAGE);
-        return;
+    final int v = Build.VERSION.SDK_INT;
+      //import android.media.MediaDataSource;
+      int codecArray[] = {
+                0 // DEFAULT
+              , MediaRecorder.AudioEncoder.AAC
+              , sdkCompat.AUDIO_ENCODER_OPUS
+              , 0 // CODEC_CAF_OPUS (specific Apple)
+              , 0 // CODEC_MP3 (not implemented)
+              , sdkCompat.AUDIO_ENCODER_VORBIS
+              , 0 // CODEC_PCM (not implemented)
+      };
+      int formatsArray[] = {
+                MediaRecorder.OutputFormat.MPEG_4   // DEFAULT
+              , MediaRecorder.OutputFormat.MPEG_4 // CODEC_AAC
+              , sdkCompat.OUTPUT_FORMAT_OGG       // CODEC_OPUS
+              , 0                                 // CODEC_CAF_OPUS (this is apple specific)
+              , 0                                 // CODEC_MP3
+              , sdkCompat.OUTPUT_FORMAT_OGG       // CODEC_VORBIS
+              , 0                                 // CODEC_PCM
+      };
+      String pathArray[] = {
+                "sound.acc"   // DEFAULT
+              , "sound.acc"   // CODEC_AAC
+              , "sound.opus"  // CODEC_OPUS
+              , "sound.caf"   // CODEC_CAF_OPUS (this is apple specific)
+              , "sound.mp3"   // CODEC_MP3
+              , "sound.ogg"   // CODEC_VORBIS
+              , "sound.wav"   // CODEC_PCM
+      };
```

Comment:
Why did you declared this inside the method? This may cause a memory leak since when `startRecorder` is being called many times and released, the garbage collector in java will also run many times. Try to provide the `const` or `static` variables outside the method.